app/models/configuration_script.rb

```diff
@@ -1,3 +1,17 @@
 class ConfigurationScript < ActiveRecord::Base
-  belongs_to :configuration_manager, :class_name => 'ManageIQ::Providers::ConfigurationManager'
+  belongs_to  :manager,
+              :class_name  => 'ManageIQ::Providers::ConfigurationManager',
+              :foreign_key => :configuration_manager_id
+  include ProviderObjectMixin
+
+  def run(vars=nil)
+    extra_vars = vars || variables
```

Comment:
I agree we should merge with defaults. Also we should add `limit` to the argument list and leave the implementation to the client gem.